guides/source/action_cable_overview.md

```diff
@@ -0,0 +1,541 @@
+
+```ruby
+# app/channels/application_cable/chat_channel.rb
+class ChatChannel < ApplicationCable::Channel
+  def subscribed
+  end
+end
+```
+
+## Client-Side Components
+### Connections
+Consumers require an instance of the connection on their side this is established via JavaScript:
+
+#### Connect Consumer
+```coffeescript
+# app/assets/javascripts/application_cable.coffee
+#= require cable
+
+@App = {}
+App.cable = Cable.createConsumer("ws://cable.example.com")
+```
+The `ws://cable.example.com address` must point to your set of Action Cable servers, and it
+must share a cookie namespace with the rest of the application (which may live under http://example.com).
+This ensures that the signed cookie will be correctly sent.
+
+#### Subscriber
+When a consumer is subscribed to a channel, they act as a subscriber. A consumer can act as a subscriber to a given channel any number of times. For example, a consumer could subscribe to multiple chat rooms at the same time. (And remember that a physical user may have multiple consumers, one per tab/device open to your connection).
+
+A consumer becomes a subscriber, by creating a subscribtion to a given channel:
+```coffeescript
+# app/assets/javascripts/cable/subscriptions/chat.coffee
+# Assumes you've already requested the right to send web notifications
+App.cable.subscriptions.create { channel: "ChatChannel", room: "Best Room" }
+
+# app/assets/javascripts/cable/subscriptions/appearance.coffee
+# Assumes you've already requested the right to send web notifications
+App.cable.subscriptions.create { channel: "AppearanceChannel" }
+```
+
+While this creates the subscription, the functionality needed to respond to received data will be described later on.
+
+## Client-Server Interactions
+
+### Streams
+Streams provide the mechanism by which channels route published content (broadcasts) to its subscribers.
+
+```ruby
+# app/channels/application_cable/chat_channel.rb
+class ChatChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from "chat_#{params[:room]}"
+  end
+end
+````
+
+### Broadcastings
+
+A broadcasting is a pub/sub link where anything transmitted by a publisher is routed directly to the channel subscribers who are streaming that named broadcasting. Each channel can be streaming zero or more broadcastings. Broadcastings are purely an online queue and time dependent; If a consumer is not streaming (subscribed to a given channel), they'll not get the broadcast should they connect later.
+
+Broadcasts are called elsewhere in your Rails application:
+```ruby
+ActionCable.server.broadcast \
+  "web_notifications_#{current_user.id}", { title: 'New things!', body: 'All the news that is fit to print' }
+```
+
+The `ActionCable.server.broadcast` call places a message in the Redis' pubsub queue under a separate broadcasting name for each user. For a user with an ID of 1, the broadcasting name would be `web_notifications_1`.
```

Comment:
The `ActionCable.server.broadcast` call places a message in the current subscription adapter (Redis by default)'s pubsub queue